homeassistant/components/remember_the_milk/__init__.py

```diff
@@ -225,19 +277,68 @@ def create_task(self, call):
         import rtmapi
 
         try:
-            task_name = call.data.get('name')
+            task_name = call.data.get(CONF_NAME)
+            hass_id = call.data.get(CONF_ID)
+            rtm_id = None
+            if hass_id is not None:
+                rtm_id = self._rtm_config.get_rtm_id(self._name, hass_id)
             result = self._rtm_api.rtm.timelines.create()
             timeline = result.timeline.value
-            self._rtm_api.rtm.tasks.add(
-                timeline=timeline, name=task_name, parse='1')
-            _LOGGER.debug('created new task "%s" in account %s',
-                          task_name, self.name)
+
+            if hass_id is None or rtm_id is None:
+                result = self._rtm_api.rtm.tasks.add(
+                    timeline=timeline, name=task_name, parse='1')
+                _LOGGER.debug('created new task "%s" in account %s',
+                              task_name, self.name)
+                self._rtm_config.set_rtm_id(self._name,
+                                            hass_id,
+                                            result.list.id,
+                                            result.list.taskseries.id,
+                                            result.list.taskseries.task.id)
+            else:
+                self._rtm_api.rtm.tasks.setName(name=task_name,
+                                                list_id=rtm_id[0],
+                                                taskseries_id=rtm_id[1],
+                                                task_id=rtm_id[2],
+                                                timeline=timeline)
+                _LOGGER.debug('updated task with id "%s" in account '
+                              '%s to name %s',
+                              hass_id, self.name, task_name)
         except rtmapi.RtmRequestFailedException as rtm_exception:
             _LOGGER.error('Error creating new Remember The Milk task for '
                           'account %s: %s', self._name, rtm_exception)
             return False
         return True
 
+    def complete_task(self, call):
+        """Complete a task that was previously created by this component."""
+        import rtmapi
+
+        hass_id = call.data.get(CONF_ID)
+        if hass_id is None:
```

Comment:
This can't happen with service schema validation.

Reply:
fixed